### PR TITLE
POST Reply

### PR DESCRIPTION
**Pull request by tayyaba17**
Replies are now succesfully posted to datastore! I have to hide some things from the user (the parentID div and tags-reply div). I also need to check out the DataServlet file before this can be merged because when there is a reply, there is an error that arises with loading that I am trying to debug. Also need to remove console.log() statements.

### CODE REVIEW

File: portfolio/src/main/webapp/script.js [hunk 1]
```diff
@@ -39,19 +39,26 @@ function defaultPage() {
...
   params.append('text-input', textEle.value);
   params.append('sender', nameEle.value);
-  params.append('tags', tagEle.value);
+  params.append('parentID', parentidEle.value);
+
+  console.log("In SendAndGet");
+  console.log("This is the text value", textEle.value);
+  console.log("This is the sender value", nameEle.value);
+  console.log("This is the tag value", tagEle.value);
+  console.log("This is the parentID value", parentidEle.value);
```
Comment (tayyaba17) [inline portfolio/src/main/webapp/script.js]: Will delete

Comment (smarquez-jpg) [inline portfolio/src/main/webapp/script.js]: Don't forget to remove the print statements before merging into master

**Review by smarquez-jpg — APPROVED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -122,12 +123,14 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
  
     String nickname = request.getParameter(SENDER_PARAMETER);
     String postTag = request.getParameter(TAG_PARAMETER);
+    String parentIDString = request.getParameter(PARENTID_PARAMETER);
+    long parentID = Long.parseLong(parentIDString);
```
Comment (peizhaoou) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:94]: You would probably want to handle the parse exception here if |parentIDString| is not a valid long integer.

Comment (peizhaoou) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:94]: For better error messages to the client side.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -122,12 +123,14 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
     if (postTag == null || postTag.isEmpty()) {
       postTag = InternalTags.defaultTag();
     }
     
     // TODO: Get parentID from client when a reply is made.
-    long parentID = 0;
+//    long parentID = 0;
```
Comment (peizhaoou) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Nit: remove?

**Review by peizhaoou — APPROVED** [review]
Mostly LGTM

**Pull request merged**